### DISCUSSION TITLE
Improving Timeout Configuration for SSE and WS Connections

### DISCUSSION THREAD

**Discussion by cl3mcg** [gowebly/gowebly#115] [Ideas]
Hello everyone,

I noticed a specific comment in the `server.go` file that explains why the `ReadTimeout` and `WriteTimeout` keys in the `Server` struct are set to 5 and 10 seconds, respectively. While this configuration is generally useful, it can interfere with Server-Sent Events (SSE) or WebSocket (WS) connections.

With the current setup, it is challenging to properly establish SSE or WS connections in the Go server because the connection resets after 10 seconds due to the `WriteTimeout` setting.
It is unclear how the `ReadTimeout` setting interfere with the SSE or WS connections, but this also has an undesirable effect as multiple "_connection reset_" errors are displayed in the web browser console.

To address this, I suggest adding further comments in the `server.go` file to explain that if SSE or WS need to be set up, it would be necessary to comment out or remove the `ReadTimeout` and `WriteTimeout` keys and their respective values.

Here is the updated code snippet with the additional comments:

`server.go`
```go
// Create a new server instance with options from environment variables.
// For more information, see https://blog.cloudflare.com/the-complete-guide-to-golang-net-http-timeouts/
server := &http.Server{
  Addr:         fmt.Sprintf(":%d", port),
  ReadTimeout:  5 * time.Second,
  WriteTimeout: 10 * time.Second,
  // Note: The ReadTimeout and WriteTimeout settings may interfere with SSE or WS connections.
  // For SSE or WS, these timeouts can cause the connection to reset after 10 seconds due to the WriteTimeout setting.
  // If you plan to use SSE or WS, consider commenting out or removing the ReadTimeout and WriteTimeout key-value pairs.
}
```

This additional context should help other developers understand the implications of the timeout settings and adjust them as needed for SSE or WS connections.

As I have a limited understanding of this specific configuration, if anyone has a better idea or another way to set up properly working SSE or WS connections without altering the `ReadTimeout` and `WriteTimeout` keys in the `Server` struct, please feel free to share it.

Thank you for your attention to this matter.

Best regards,

**Comment by koddr** [comment]
@cl3mcg thanks!

Could you please do a PR with these changes?

**Reply by cl3mcg** [reply]
Hello and thank you for your response.
I will try to do so. 😊
